### PR TITLE
Setup Core Structure

### PR DESCRIPTION
**Pull request by cptpiepmatz**

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (dr4hcu5-jan) [inline .github/workflows/ci.yml:1]: Maybe move the documentation build job into a separate pipeline to allow for a manual documentation build

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+RUN apt update -y && apt upgrade -y
+WORKDIR /app
+
+# install volta directly from source
+ARG VOLTA_TAG
+RUN cargo install --git https://github.com/volta-cli/volta.git --tag $VOLTA_TAG --locked
+ENV PATH="bin/volta/bin:$PATH"
+
+# install node
+COPY package.json .
```
Comment (dr4hcu5-jan) [inline Dockerfile:16]: Use `COPY --link` to allow caching the `package.json` file if a previous step might invalidate them.

https://docs.docker.com/reference/dockerfile/#benefits-of-using---link

```suggestion
COPY --link package.json .
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+RUN volta run node --version
+
+# install dependencies
+COPY package-lock.json .
```
Comment (dr4hcu5-jan) [inline Dockerfile:20]: Same here, use `COPY --link`

```suggestion
COPY --link package-lock.json .
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+RUN volta run npm ci
+
+# build the app
+COPY . .
```
Comment (dr4hcu5-jan) [inline Dockerfile:24]: Same here, use `COPY --link`

```suggestion
COPY --link . .
```

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+RUN volta run npm run build
+
+
+# build a file server to use for hosting the static files
+FROM rust:alpine AS build-server
+RUN apk update && apk upgrade
```
Comment (dr4hcu5-jan) [inline Dockerfile:30]: Decrease Build Stage Size by disabling package cache
```suggestion
RUN apk upgrade --no-cache
```

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+RUN apk add --no-cache musl-dev build-base
+
+ENV CARGO_HOME=/cargo
+ARG DUFS_VERSION
+RUN cargo install dufs --version $DUFS_VERSION --locked
+
+
+# host the app via the static file server
+FROM alpine:latest AS app 
+RUN apk update && apk upgrade
```
Comment (dr4hcu5-jan) [inline Dockerfile:40]: Disable package cache to decrease output image size
 
```suggestion
RUN apk upgrade --no-cache
```

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+WORKDIR /wisdom-oss
+COPY --from=build-app /app/dist/wisdom-oss/frontend/browser /wisdom-oss/app
```
Comment (dr4hcu5-jan) [inline Dockerfile:42]:
```suggestion
COPY --link --from=build-app /app/dist/wisdom-oss/frontend/browser /wisdom-oss/app
```

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+COPY --from=build-server /cargo/bin/dufs /wisdom-oss/
```
Comment (dr4hcu5-jan) [inline Dockerfile:43]:
```suggestion
COPY --link --from=build-server /cargo/bin/dufs /wisdom-oss/
```

**Review by dr4hcu5-jan — CHANGES_REQUESTED** [review]
Initially looks good, but visual inspection will follow until Sunday. However, the requested changes are not dependent of this

File: .github/workflows/ci.yml
Comment (cptpiepmatz) [reply .github/workflows/ci.yml:1]: Could be useful in the future, right now it just checks if it is possible to build documentation

**Comment by dr4hcu5-jan** [comment]
@cptpiepmatz Please add an README.md with some content to the repository

File: src/core/navbar/navbar-user/navbar-user.component.html [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    >
+      <span
+        class="icon"
+        [class.has-text-primary]="rememberLogin()"
+        [class.has-opacity-60]="!rememberLogin()"
+        (click)="toggleRemember()"
+      >
+        <ng-icon size="1.6em" name="remixRotateLockFill"></ng-icon>
+      </span>
+    </button>
```
Comment (dr4hcu5-jan) [inline src/core/navbar/navbar-user/navbar-user.component.html:24]: The tooltip text is not fully visible on desktop clients with a width ≤ 1920px

Bildschirmfoto am 2024-12-13 um 09 18 40

Comment (dr4hcu5-jan) [reply src/core/navbar/navbar-user/navbar-user.component.html:24]: Furthermore, the tooltip text is not available in German

File: src/core/navbar/navbar-user/navbar-user.component.html [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+          <span
+            *ngIf="userDetails()!.administrator"
+            class="icon has-tooltip-top"
+            data-tooltip="Administrator"
+          >
+            <ng-icon name="remixVipCrown2Fill"></ng-icon>
+          </span>
+          <span style="white-space: nowrap">
+            {{ userDetails()!.name }}
+          </span>
```
Comment (dr4hcu5-jan) [inline src/core/navbar/navbar-user/navbar-user.component.html:55]: The administrator crown icon shifts the users full name to the right which looks wrong in comparison to a non-admin user

**Comment by dr4hcu5-jan** [comment]
There also is a content shift which does not correlate with the increased layout width when making the viewport 1921px wide

**Viewport Width = 1921px**
Bildschirmfoto am 2024-12-13 um 09 27 51

**Viewport Width = 1920px**
Bildschirmfoto am 2024-12-13 um 09 27 46

File: .github/workflows/ci.yml
Comment (dr4hcu5-jan) [reply .github/workflows/ci.yml:1]: Okay, maybe insert a comment for this job explaining it, but not required

File: src/core/navbar/navbar-user/navbar-user.component.html (hunk 8 above)
Comment (cptpiepmatz) [reply src/core/navbar/navbar-user/navbar-user.component.html]: The tooltip is translated now, see 2472f160249f4e799778ecc57681df68a025fddd
The tooltip is now also right aligned with some padding to look more floaty, see a9888091a2b38f7474be290df4071b6f8d1a88bd

File: src/core/navbar/navbar-user/navbar-user.component.html (hunk 9 above)
Comment (cptpiepmatz) [reply src/core/navbar/navbar-user/navbar-user.component.html:55]: image
I think this looks worse

Comment (dr4hcu5-jan) [reply src/core/navbar/navbar-user/navbar-user.component.html:55]: Hmm, also true, maybe we can find another way later on (e.g. placing that crown over the profile picture in one of the lower corners)

**Review by dr4hcu5-jan — APPROVED** [review]
Looks good now

**Review by dr4hcu5-jan — CHANGES_REQUESTED** [review]
Block until readme is uploaded

**Comment by cptpiepmatz** [comment]
I added a readme, most technical stuff is not mentioned in it as the CONTRIBUTING file should handle that

**Review by dr4hcu5-jan — APPROVED** [review]

**Pull request merged**